discovery_engine/lib/src/ffi/types/document/document_vec.dart

```diff
@@ -65,4 +68,11 @@ extension DocumentSliceFfi on List<DocumentFfi> {
     ffi.drop_document_vec(boxedVec);
     return res;
   }
+
+  List<Document> toDocumentList() {
+    return asMap()
+        .entries
+        .map((e) => e.value.toDocument(batchIndex: e.key))
+        .toList();
+  }
```

Comment:
since this is a one-liner anyways, it could just be written as
```suggestion
  List<Document> toDocumentList() => asMap()
      .entries
      .map((e) => e.value.toDocument(batchIndex: e.key))
      .toList();
```